tools/Import/Importer.cs

```diff
@@ -0,0 +1,271 @@
+
+            if (string.IsNullOrEmpty(fhirEndpoint2))
+            {
+                fhirEndpoint2 = FhirEndpoint;
+            }
+
+            var globalLogPrefix = $"GlobalBlobRange=[{NumberOfBlobsToSkip + 1}-{NumberOfBlobsToImport}]";
+            Console.WriteLine($"{globalLogPrefix}: Starting...");
+            var blobContainerClient = GetContainer(ConnectionString, ContainerName);
+            var blobs = blobContainerClient.GetBlobs().Skip(NumberOfBlobsToSkip).Take(NumberOfBlobsToImport - NumberOfBlobsToSkip).ToList();
+            var writers = 0L;
+            var epCalls = 0L;
+            var waits = 0L;
+            var globalResourceCount = 0L;
+            var swWrites = Stopwatch.StartNew();
+            var swReport = Stopwatch.StartNew();
+            var swReportReads = Stopwatch.StartNew();
+            BatchExtensions.ExecuteInParallelBatches(blobs, ReadThreads, BlobRangeSize, (reader, blobRangeInt) =>
+            {
+                var localSw = Stopwatch.StartNew();
+                var localResourceCount = 0L;
+                var blobRangeIndex = blobRangeInt.Item1;
+                var blobRange = blobRangeInt.Item2;
+                var localLogPrefix = $"Reader={reader}.BlobRange=[{NumberOfBlobsToSkip + (blobRangeIndex * BlobRangeSize) + 1}-{NumberOfBlobsToSkip + (blobRangeIndex * BlobRangeSize) + blobRange.Count}]";
+                Console.WriteLine($"{localLogPrefix}: Starting...");
+
+                BatchExtensions.ExecuteInParallelBatches(GetLinesInBlobRange(blobRange, localLogPrefix), WriteThreads / ReadThreads, 100, (thread, lineBatch) =>
+                {
+                    Interlocked.Increment(ref writers);
+                    foreach (var line in lineBatch.Item2)
+                    {
+                        Interlocked.Increment(ref globalResourceCount);
+                        Interlocked.Increment(ref localResourceCount);
+                        PutResource(line, ref localResourceCount, ref epCalls, ref waits);
+                        if (swReport.Elapsed.TotalSeconds > ReportingPeriodSec)
+                        {
+                            lock (swReport)
+                            {
+                                if (swReport.Elapsed.TotalSeconds > ReportingPeriodSec)
+                                {
+                                    var resWritten = Interlocked.Read(ref globalResourceCount);
+                                    var resRead = Interlocked.Read(ref resourcesRead);
+                                    Console.WriteLine($"{globalLogPrefix}.Readers={Interlocked.Read(ref readers)}.Writers={Interlocked.Read(ref writers)}.EndPointCalls={Interlocked.Read(ref epCalls)}.Waits={Interlocked.Read(ref waits)}: reads={resRead} writes={resWritten} secs={(int)swWrites.Elapsed.TotalSeconds} read-speed={(int)(resRead / swReads.Elapsed.TotalSeconds)} lines/sec write-speed={(int)(resWritten / swWrites.Elapsed.TotalSeconds)} res/sec");
```

Comment:
The reporting feature is really nice. I'd replace globalLogPrefix with localLogPrefix and add "..." to add more info and improve readability on the screen.
Console.WriteLine($"... {localLogPrefix}.Readers={Interlocked.Read(ref readers)}.Writers={Interlocked.Read(ref writers)}.EndPointCalls={Interlocked.Read(ref epCalls)}.Waits={Interlocked.Read(ref waits)}: reads={resRead} writes={resWritten} secs={(int)swWrites.Elapsed.TotalSeconds} read-speed={(int)(resRead / swReads.Elapsed.TotalSeconds)} lines/sec write-speed={(int)(resWritten / swWrites.Elapsed.TotalSeconds)} res/sec");

Reply:
Logs are cluttered with lots of retry messages, so it is not easy to navigate without pattern searches anyway. I don't think adding "..." would be that useful.